### PR TITLE
BSM-142: Renamed MostCritical to HighestSeverity

### PR DESCRIPTION
**Pull request by fooker**
The class was renamed to reflect the renaming request as the user output
is equal to the class name.

JIRA: http://issues.opennms.org/browse/BSM-142
Bamboo: http://bamboo.internal.opennms.com:8085/browse/OPENNMS-ONMS477

### CODE REVIEW

**Comment by mvrueden** [comment]
The update for the administration documentation is missing. See http://docs.opennms.org/opennms/branches/features-bsm/guide-admin/guide-admin.html#_calculation_of_the_operational_status "Reduce Functions".

The update for the developers documentation is missing. See http://docs.opennms.org/opennms/branches/features-bsm/guide-development/guide-development.html#_business_service_monitoring.

The update on the wiki page is missing. See http://www.opennms.org/wiki/BusinessServiceMonitoring#State_Calculation

File: features/bsm/persistence/api/src/main/java/org/opennms/netmgt/bsm/persistence/api/functions/reduce/HighestSeverityEntity.java [hunk 1]
```diff
@@ -33,6 +33,6 @@
 
 @Entity
 @DiscriminatorValue(value="most-critical")
```
Comment (mvrueden) [inline features/bsm/persistence/api/src/main/java/org/opennms/netmgt/bsm/persistence/api/functions/reduce/HighestSeverityEntity.java]: I'd say lets also change this to `higehst-severity`.

**Comment by fooker** [comment]
Good catch. Fixed.

File: features/bsm/persistence/api/src/main/java/org/opennms/netmgt/bsm/persistence/api/functions/reduce/HighestSeverityEntity.java (hunk 1 above)
Comment (fooker) [inline features/bsm/persistence/api/src/main/java/org/opennms/netmgt/bsm/persistence/api/functions/reduce/HighestSeverityEntity.java]: Fixed.

**Comment by mvrueden** [comment]
Please also add a changelog.xml entry to update all `most-critical` entries to `highest-severity`.

**Comment by mvrueden** [comment]
Hey @fooker, I tried this, but it seems there is a problem with the provided changelog.xml file.
I get the following error: `Error executing SQL UPDATE bsm_reduce SET type = 'highest-severity' WHERE most-critical: ERROR: column "most" does not exist`

**Comment by mvrueden** [comment]
As soon as Bamboo goes green, let's merge this.

**Pull request merged**